docs/how_jstz_works.md

```diff
@@ -0,0 +1,60 @@
+---
+title: 👨‍⚖️ How jstz works?
+---
+
+<script setup>
+import VPButton from "vitepress/dist/client/theme-default/components/VPButton.vue";
+</script>
+
+# 👨‍⚖️ `How jstz works?`
+
+## Overview
+
+`jstz` is a JavaScript/Typescript server runtime for Tezos 2.0 designed to provide a great developer experience by aiming to be compatible with web conventions.
+
+Through `jstz` developers can set up, deploy and test so called _smart functions_ written in Javascript/Typescript that can get directly executed on Tezos L2.
+
+Jstz provides a local sandboxed environment for developers to test the functions without deploying them to L2. It provides a simple interface through which one can deploy smart functions and then call them by sending HTTP requests to a particular _smart function address_.
+
+## How it works?
+
+Jstz has to ensure that smart functions written in JS/TS can compile directly to WASM - the assembly used for writing Smart Rollups. Therefore 'jstz' is built on _Boa_ - a Javascript engine written in Rust.
+
+In its core (jstz_core), Jstz uses Boa and enables Rust types to be passed around as JavaScript objects. This allows implementation and registration of various APIs written in Rust and their usage as if they were native Javascript objects.
+
+The APIs (implemented in jstz_api) enable interaction with the storage of a rollup kernel (a key-value store) and also implement Request and Response APIs based on the Fetch API specification enabling calling of other smart functions.
+
+##
+
+### KV store
+
+When writing smart functions, we need a way to store data across different calls of the functions. Therefore, jstz _smart functions_ implement a persistent key-value store used for storing and retrieval of arbitrary JSON blobs. This store can be accessed through a global _Kv_ object.
+
+The key-value store implements _optimistic concurrency control scheme_. It is optimistically assumed that conflicts between different transactions (snapshots of the persistent kv store) are sufficiently rare thus not interfering each other. Before commiting, the transaction verifies whether no other transaction has modified the data it has read.
+
+The transactions performed over the KV store offer ACID guarantees and serializability, therefore any transaction can be commited only if it does not conflict with any previously commited ones.
+
+In each transaction, the repeated access to the same key is optimized through caching. Similarly, writes are buffered until the transaction is commited at which point it gets flushed to the persistent KV storage.
```

Comment:
Might be good to move these paragraphs into the paragraph mentioning `jstz_core`. And then mention that the `Kv` store is implemented on top of `jstz_core::kv`.